src/components/DotNavigation/DotNavigation.tsx

```diff
@@ -0,0 +1,294 @@
+};
+
+const getFiveFromStart = (start: number) => [
+  start,
+  start + 1,
+  start + 2,
+  start + 3,
+  start + 4,
+];
+
+export class DotNavigation extends React.Component<
+  DotNavigationProps,
+  DotNavigationState
+> {
+  state = {
+    savedCurrentIndex: 0,
+    animation: null,
+    longListKey: 0,
+  };
+
+  static displayName = 'DotNavigation';
+
+  static defaultProps: DotNavigationProps = {
+    currentIndex: 0,
+    length: MAX_SHORT_LIST_LENGTH,
+    onSelect(index: number) {},
+    showBorder: false,
+    theme: Theme.Dark,
+    'aria-label': 'Dot Navigation',
+  };
+
+  static shouldAnimate = (
+    currentIndex: number,
+    savedCurrentIndex: number,
+    length: number,
+  ) =>
+    (currentIndex > 2 && currentIndex < length - 3) ||
+    (currentIndex === 2 && currentIndex < savedCurrentIndex) ||
+    (currentIndex === length - 3 && currentIndex > savedCurrentIndex);
+
+  static hasCurrentIndexChanged = (
+    currentIndex: number,
+    savedCurrentIndex: number,
+    length: number,
+  ) =>
+    isNumber(currentIndex) &&
+    currentIndex >= 0 &&
+    currentIndex < length &&
+    currentIndex !== savedCurrentIndex;
+
+  static getLongListAnimation = (
+    nextProps: DotNavigationProps,
+    prevState: DotNavigationState,
+  ) => {
+    const { currentIndex, length } = nextProps;
+    const { savedCurrentIndex, longListKey } = prevState;
+
+    return DotNavigation.hasCurrentIndexChanged(
+      currentIndex,
+      savedCurrentIndex,
+      length,
+    )
+      ? {
+          ...prevState,
+          savedCurrentIndex: currentIndex,
+          ...(DotNavigation.shouldAnimate(
+            currentIndex,
+            savedCurrentIndex,
+            length,
+          ) && {
+            animation:
+              currentIndex < savedCurrentIndex
+                ? Animation.Back
+                : Animation.Forward,
+            longListKey: getUniqueLongListKey(longListKey),
+          }),
+        }
+      : null;
+  };
+
+  static getDerivedStateFromProps = (
+    nextProps: DotNavigationProps,
+    prevState: DotNavigationState,
+  ) =>
+    nextProps.length <= MAX_SHORT_LIST_LENGTH
+      ? null
+      : DotNavigation.getLongListAnimation(nextProps, prevState);
+
+  callOnSelect = (index: number) => () => this.props.onSelect(index);
+
+  getRadio = ({ key, checked, size, fakeRadio }: RadioProps) => (
+    <div className={classNames(styles.radio)}>
+      <div
+        key={key}
+        className={classNames(styles.dot, {
+          [styles.small]: size === Size.Small,
+          [styles.tiny]: size === Size.Tiny,
+          [styles.bordered]: this.props.showBorder === true,
+          [styles.light]: this.props.theme === Theme.Light,
+          [styles.checked]: checked,
+        })}
+        data-hook={fakeRadio ? undefined : DotNavigationDataHooks.Dot}
+      />
+    </div>
+  );
+
+  renderFakeRadio = (key: string) =>
+    this.getRadio({ key, checked: false, size: Size.Tiny, fakeRadio: true });
```

Comment:
I added tabIndex={-1} and aria-hidden to HTML element, which represents a fake dot. There is no need to change <code>wix-ui-core</code> for this as I don't use their <code>RadioButton</code> for fake dots.

I tried though. The problem was that <code>RadioButton</code> had an optional prop aria-label, which I didn't pass because the dots were fake. But accessibility tests failed because this prop was absent. So I decided not to use this component at all. To be honest, we don't need any of its functionality for fake dots as they are only for visual animation purposes.